File: C++/162-Find Peak Element.cpp
// Runtime: 8 ms, faster than 80.79% of C++ online submissions for Find Peak Element.
// Memory Usage: 8.6 MB, less than 54.60% of C++ online submissions for Find Peak Element.
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int left = 0, right = nums.size() - 1;
        while (left < right) {
            int mid = (left + right) / 2;
            if (nums[mid] > nums[mid + 1])
                right = mid;
            else
                left = mid + 1;
        }
        return left;
    }
};

File: Python3.x/162-Find Peak Element.py
# Runtime: 32 ms, faster than 94.17% of Python3 online submissions for Find Peak Element.
# Memory Usage: 13.3 MB, less than 33.43% of Python3 online submissions for Find Peak Element
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        max_idx = 0
        for i, v in enumerate(nums):
            if v > nums[max_idx]: max_idx = i
        return max_idx
